pkg/agent/controller/egress/egress_controller.go

```diff
@@ -851,6 +878,10 @@ func (c *EgressController) syncEgress(egressName string) error {
 		if err := c.ipAssigner.AssignIP(desiredEgressIP, egress.Status.EgressNode != c.nodeName); err != nil {
 			return err
 		}
+		_, exists := egressIPMap[desiredEgressIP]
```

Comment:
I think you should make `AssignIP` return a bool to indicate whether the IP was assigned before the operation.

Reply:
@tnqn `AssignIP` approach won't work because when antrea agent is getting restarted it again assigns the IP to the interface, and this condition `if a.assignedIPs.Has(ip)` returns false. I have tested this locally.